reconcile/utils/terraform_client.py

```diff
@@ -155,26 +155,85 @@ def plan(self, enable_deletion):
     def terraform_plan(
         self, plan_spec: dict, enable_deletion: bool
     ) -> tuple[bool, list[AccountUser], bool]:
+        disabled_deletion_detected = False
         name = plan_spec["name"]
         tf = plan_spec["tf"]
         return_code, stdout, stderr = tf.plan(
             detailed_exitcode=False, parallelism=self.parallelism, out=name
         )
         error = self.check_output(name, "plan", return_code, stdout, stderr)
-        disabled_deletion_detected, created_users = self.log_plan_diff(
-            name, tf, enable_deletion
-        )
+        plan = self._get_json_plan(name, tf)
+
+        resource_changes = plan.get("resource_changes")
+        if resource_changes is not None:
+            disabled_deletion_detected = self._detect_disabled_deletion(
```

Comment:
will `disabled_deletion_detected` not jump between being True and False?

Reply:
Right, that would be expected in case a disabled deletion is detected. `disabled_deletion_detected` is assigned two times, first during initialization providing the default value (line 158) and then in this line where we get the value from `_detect_disabled_deletion`. 
Did I get the question right?

Reply:
@maorfr can you please elaborate? I don't understand your comment :question:

Reply:
once `disabled_deletion_detected` has been set to True, it should never be False again.

Reply:
Got it. That's the case here. False is only assigned once in line 158/201 to declare the variable.

Reply:
but `_detect_disabled_deletion` can return either True or False, and you are setting the "global" variable...
am i just not seeing it?

Reply:
after discussing, the conclusion is that since there is no loop anymore, there is no risk.